### ISSUE TITLE
Invalid arguments to shell/command/raw cause exception.

### ISSUE THREAD

**Issue by pcastellazzi** [willthames/ansible-lint#141]
You can trigger the error with the current version from git (f34d094) and this playbook:

```yaml
---
- hosts: all
  gather_facts: no
  tasks:
    - shell: PATH=$PATH:/usr/local/bin/pip
```

My solution:

```diff
diff --git a/lib/ansiblelint/rules/CommandsInsteadOfArgumentsRule.py b/lib/ansiblelint/rules/CommandsInsteadOfArgumentsRule.py
index 2161729..9161dc6 100644
--- a/lib/ansiblelint/rules/CommandsInsteadOfArgumentsRule.py
+++ b/lib/ansiblelint/rules/CommandsInsteadOfArgumentsRule.py
@@ -40,7 +40,7 @@ class CommandsInsteadOfArgumentsRule(AnsibleLintRule):
                   'rmdir': 'state=absent', 'rm': 'state=absent'}

     def matchtask(self, file, task):
-        if task["action"]["module"] in self._commands:
+        if task["action"]["module"] in self._commands and task["action"]["module_arguments"]:
             executable = os.path.basename(task["action"]["module_arguments"][0])
             if executable in self._arguments and \
                     boolean(task['action'].get('warn', True)) and \
diff --git a/lib/ansiblelint/rules/CommandsInsteadOfModulesRule.py b/lib/ansiblelint/rules/CommandsInsteadOfModulesRule.py
index 30b0fa5..671249b 100644
--- a/lib/ansiblelint/rules/CommandsInsteadOfModulesRule.py
+++ b/lib/ansiblelint/rules/CommandsInsteadOfModulesRule.py
@@ -41,7 +41,7 @@ class CommandsInsteadOfModulesRule(AnsibleLintRule):
                 'unzip': 'unarchive', 'tar': 'unarchive'}

     def matchtask(self, file, task):
-        if task["action"]["module"] in self._commands:
+        if task["action"]["module"] in self._commands and task["action"]["module_arguments"]:
             executable = os.path.basename(task["action"]["module_arguments"][0])
             if executable in self._modules and \
                     boolean(task['action'].get('warn', True)) and \
```

**Comment by willthames** [comment]
Might be worth raising a pull request for this - while I'm happy to take the patch, I'd rather take patches through pull requests than through issues. If you have any concerns about that, let me know.

**Comment by willthames** [comment]
This change just hides the fact that the task doesn't actually do anything - setting environment should be done through the `environment` key. ansible-lint shouldn't stack trace though.

**Comment by pcastellazzi** [comment]
A pull request has been made for this issue #144.
The PR aims to fix the bug, not to report the NOOP to the user. I think a new rule will be needed to handle the specifics.

**Comment by willthames** [comment]
Fixed with #144
